src/ckeditor.tsx

```diff
@@ -440,6 +452,7 @@ export interface Props<TEditor extends Editor> extends InferProps<typeof CKEdito
 		EditorWatchdog: typeof EditorWatchdog;
 		ContextWatchdog: typeof ContextWatchdog;
 	};
+	context?: CKEditorConfigContextMetadata;
```

Comment:
What is the purpose of introducing a new property that accepts arbitrary data?

I feel like this integration already deals with too many things called `context` and a new one will only make things worse. Wouldn't adding a new optional property called `id` or `name` be sufficient and easier to document?

Reply:
I was struggling with this too. `id` cannot be used because it is used to decide if the editor should be reloaded or not. So I'm nearly 100% sure that some of the integrations do something like this: `id={JSON.stringify(config)}` which is not we want in editors, mam. The `name` on the other hand can be misleading with HTML `name` attribute. The main concept of `context` property is to:

1. Tell the user that it's only context-only configuration, so it'll be ignored if the component is not used within context.
2. Pass more additional fields from child to parent. In other words, we can add some metadata about placement of the editor to context parent. Like this:

```tsx
<CKEditorContext
  context={ ClassicEditor.Context }
  contextWatchdog={ ClassicEditor.ContextWatchdog }
  onChangeEditorsMap={ editors => {
    console.log( editors.editor2.stuff );
  }}
>
  <CKEditor
    editor={ ClassicEditor }
    data="<h2>Another Editor</h2><p>... in common Context</p>"
    context={{
      editorName: 'editor2',
      stuff: 2
    }}
  />
</CKEditorContext>
```

`Stuff` is forwarded to parent and can be used to forward some `CKEditor` state to parent using our context.